### ISSUE TITLE
[Enhancement] Add Conventional Commits

### ISSUE THREAD

**Issue by ltribolet** [ltribolet/Lychee#60]
Adding this https://github.com/ramsey/conventional-commits

So we would follow https://www.conventionalcommits.org/en/v1.0.0/

But it would mean dropping Gitmoji I think.

**Comment by sanghin** [comment]
Do you have something like [semantic-release](https://github.com/semantic-release/semantic-release) on PHP? If so having formatted commit message and a commit linter would be necessary

**Comment by ltribolet** [comment]
Those are 2 different things. This is for commit message formatting enforcement. While Semantic Release appears to build the release package. While I understand that Semantic release is based on commit messages, I see those 2 being different. I am not aware of such tool dedicated to PHP.

**Comment by sanghin** [comment]
Yes they are different tools but if a semantic-release like module exist on php, you will need to have a commit message enforcement tool so you can generate changelog reliably, that was my point 😄

**Comment by ltribolet** [comment]
This is not something planned at the moment. But I'm not sure a tool like Conventional Commits and Semantic Release would be compatible for instance.
